ragaai_catalyst/tracers/tracer.py

```
@@ -775,7 +810,15 @@ def _setup_agentic_tracer(self, instrumentors):
         # Instrument all specified instrumentors
         for instrumentor_class, args in instrumentors:
-            instrumentor_class().instrument(tracer_provider=tracer_provider, *args)
+            # Create an instance of the instrumentor
+            instrumentor = instrumentor_class()
+            
+            # Uninstrument only if it is already instrumented
+            if isinstance(instrumentor, LangChainInstrumentor) and instrumentor._is_instrumented_by_opentelemetry:
+                instrumentor.uninstrument()
+            
+            # Instrument with the provided tracer provider and arguments
+            instrumentor.instrument(tracer_provider=tracer_provider, *args)
```

Ruff (0.8.2)

Star-arg unpacking after a keyword argument is strongly discouraged

(B026)